Define dependency maintenance policy (incl. Spring)

npepinpe opened the pull request:
## Description

This PR defines our basic support policy for the Camunda Orchestration Cluster, and the specific way we handle Spring's shorter EOL cycle.

I wasn't sure where this should go since we don't really mention anything about this anywhere, _and_ we don't have anything about our own support policy. Please suggest otherwise.

Additionally, it would have been good to add some timeline visualization to show the difference between our support policy and Spring's, but I am terrible at this, so if you have some idea, go for it.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [x] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (create draft PR and/or add `hold` label)
- [ ] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [x] My changes are for **an upcoming minor release** and:
  - [x] are in the `/docs` directory (version 8.8).
  - [ ] are in the `/versioned_docs/version-8.7/` directory (version 8.7).
- [ ] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [ ] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [ ] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

npepinpe commented:
Remaining to do:

- [ ] Backport to versioned_docs. I'm waiting until the "main" part is reviewed before adding it to older versions, as it will pretty much be copy paste.
- [ ] I'll grab someone from engineering to also review this in terms of contents.

akeller commented:
> I wasn't sure where this should go since we don't really mention anything about this anywhere, and we don't have anything about our own support policy. Please suggest otherwise.

> Additionally, it would have been good to add some timeline visualization to show the difference between our support policy and Spring's, but I am terrible at this, so if you have some idea, go for it. 

@npepinpe can you bring this into Slack and start a thread on #ask-c8-documentation? Good discussion topics! 

Since the Orchestration cluster doesn't exist until 8.8, we have plenty of time to discuss and decide what makes sense to communicate this and where.

npepinpe commented:
Note that this affects all supported versions, I just haven't written it yet because I'm waiting to have the `main` part done (since it will be mostly similar).

christinaausley commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+
+### Spring
+
+Many server side components, such as the **Camunda Orchestration Cluster**, are Spring Boot applications, and leverage Spring Boot (and the wider Spring ecosystem) to execute fundamental functionality such as application configuration, REST infrastructure (including security), production ready features, etc.
```
This is a good place to link to the Camunda Orchestration Cluster, as defined in the glossary. We should likely make it very clear how we have defined this for both current and new users, but I just realized we haven't added this to the glossary yet.

I would propose we provide a link to this glossary definition in every instance of Camunda Orchestration Cluster for now -- additionally, is this the casing we want to standardize on? CC @MaxTru @akeller 

@npepinpe This is out of scope for this PR, so you can disregard.

akeller replied:
We are still working on some of the terminology in 8.8, but I'm pretty confident this will just be "Orchestration cluster" with no "Camunda" qualifier on the front. I definitely think it's a good opportunity in 8.8 with the intro of the Orchestration cluster that we link to the glossary definition.

akeller replied:
Related - https://github.com/camunda/camunda-docs/pull/5537

christinaausley commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+
+Spring, however, has a shorter maintenance window than Camunda for its OSS offering: versions are only supported for 12 months, versus Camunda's 18 months. To circumvent this, server side components of Camunda 8 patch releases also update Spring (and Spring Boot) minor versions, such that the latest patch release of these components uses a supported Spring version.
+
+As for libraries and SDKs meant to be included in third-party applications, Camunda follows a best effort policy to balance compatibility and securing Spring-dependent libraries.
```
```suggestion
As for libraries and SDKs meant to be included in third-party applications, Camunda follows a best effort policy to balance compatibility and secure Spring-dependent libraries.
```

christinaausley commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+
+## Dependency maintenance policies
+
+Camunda provides [a standard support policy](https://camunda.com/release-policy/) of 18 months for a particular minor version, from the date it is released.
```
```suggestion
Camunda provides [a standard support policy](https://camunda.com/release-policy/) of 18 months for a particular minor version from the date it is released.
```

christinaausley commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+## Dependency maintenance policies
+
+Camunda provides [a standard support policy](https://camunda.com/release-policy/) of 18 months for a particular minor version, from the date it is released.
+During this time, patches are regularly released which contain security and bug fixes, some of which may come from dependency updates. As such, for the
```
```suggestion
During this time, patches are regularly released containing security and bug fixes, some of which may come from dependency updates. Therefore, the
```

christinaausley commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+During this time, patches are regularly released which contain security and bug fixes, some of which may come from dependency updates. As such, for the
+vast majority of dependencies, Camunda _only_ applies patch updates.
+
+Certain dependencies used by Camunda 8, however, may have a shorter maintenance policy than Camunda itself. For those dependencies which are integral to its functionality, Camunda may adopt a different update policy, as listed below.
```
```suggestion
However, certain dependencies used by Camunda 8 may have a shorter maintenance policy than Camunda itself. Camunda may adopt a different update policy for these dependencies, as listed below.
```

npepinpe commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+
+Many server side components, such as the **Camunda Orchestration Cluster**, are Spring Boot applications, and leverage Spring Boot (and the wider Spring ecosystem) to execute fundamental functionality such as application configuration, REST infrastructure (including security), production ready features, etc.
+
+Spring, however, has a shorter maintenance window than Camunda for its OSS offering: versions are only supported for 12 months, versus Camunda's 18 months. To circumvent this, server side components of Camunda 8 patch releases also update Spring (and Spring Boot) minor versions, such that the latest patch release of these components uses a supported Spring version.
```
I applied the suggestion anyway, but changed the OSS definition.

npepinpe commented:
Backlog issue is fine. I was thinking of an easy timeline visual between patch and minor versions, but really, design is not my strong suit :)

npepinpe commented:
@christinaausley I took the latest version after applying feedback and added it to 8.7, and replaced the reference to the orchestration cluster to the separate component. Let me know if that's fine, and if so, I'll copy paste to older versions.

ThorbenLindhauer commented on docs/reference/supported-environments.md:
```diff
@@ -156,3 +156,19 @@ You can also use newer versions of Desktop and Web Modeler with older versions o
+
+Many server side components, such as the **Camunda Orchestration Cluster**, are Spring Boot applications, and leverage Spring Boot (and the wider Spring ecosystem) to execute fundamental functionality such as application configuration, REST infrastructure (including security), production ready features, etc.
+
+However, Spring has a shorter maintenance window than Camunda for its open-source software (OSS) offering. Versions are only supported for 12 months, versus Camunda's 18 months. To circumvent this, server-side components of Camunda 8 patch releases also update Spring (and Spring Boot) minor versions, such that the latest patch release of these components uses a supported Spring version.
```
You could link here to the support policy documentation for [Spring](https://spring.io/projects/spring-framework#support) and [Spring Boot](https://spring.io/projects/spring-boot#support). In that sense, 12 months for Spring is not fully accurate (it's Spring Boot, Spring Framework tends to have longer support durations). Maybe you can clarify that or make the statement more generic (like "Spring and Spring Boot minor versions can go out of maintenance during the maintenance window of a Camunda minor release").

sbuettner replied:
@npepinpe Can we explicitly limit it to the orchestration cluster?

npepinpe replied:
Spring support is a little hard to explain, since it's tied to the Spring Boot version. But I can keep it to Spring Boot for now.

> All project support dates are mapped to the version of Spring Boot that either manages it or it depends upon (the support dates for a given project will match those of the Spring Boot version it’s mapped to).

Linking is a good idea :+1:

ThorbenLindhauer reviewed (approved):
Looks fine to me. One more non-critical suggestion.

christinaausley reviewed (approved):
Thanks for resolving all of my comments!

I am approving, but did we ever resolve the conversation in [Slack](https://camunda.slack.com/archives/C026U8GBNSW/p1744130116004229) regarding placement?

I don't think this needs to block this PR, but if we do not have a definitive answer on this we should probably create a backlog issue in the docs space to revisit.

npepinpe commented:
I'd appreciate it if we could merge as soon as possible, so we can share the staged docs with the customer :pray:
